homeassistant/components/matter/adapter.py

```diff
@@ -185,6 +187,23 @@ def _create_device_registry(
             endpoint,
         )
         identifiers = {(DOMAIN, f"{ID_TYPE_DEVICE_ID}_{node_device_id}")}
+
+        network_interfaces: list[GeneralDiagnostics.Structs.NetworkInterface] = (
+            endpoint.get_attribute_value(
+                None, GeneralDiagnostics.Attributes.NetworkInterfaces
+            )
+            or []
+        )
+
+        connections = {
+            (
+                dr.CONNECTION_NETWORK_MAC,
+                convert_mac_address(network_interface.hardwareAddress),
```

Comment:
we need to guard for invalid data here as we have seen devices in the wild with malformed data in the hardwareAddress field. So my proposal is to convert this entire block into a small helper util which guards for invalid mac addresses (or the convert_mac_address util erroring)

Reply:
Thanks for the review. 

I'll look into it shortly.

Reply:
Hey, 

I looked into it and maybe there is more to it…
Maybe you can elaborate on those _malformed addresses_ and what the concern is here.

Since the matter lib will always return `bytes` for the `hardwareAddress`, a malformed mac address is not a problem, and you only have to check for excessive size. 

But that leads me to another thing, namely: _What is a valid mac address?_

Looking at:
https://github.com/home-assistant/core/blob/a5c9aa44bd8cbc648679d184011b40bb2eea2273/homeassistant/helpers/device_registry.py#L420-L437

It assumes a 48bit address. But IEEE 802.15.4 (or any other new standard) uses EUI-64, 64bit addresses, which is `CONNECTION_ZIGBEE = "zigbee"` in HA, although this is not exclusive to ZigBee and a device should either have a EUI-48 (traditional MAC) or a EUI-64. So there isn't really a need to differentiate between the two. 

There is also the `CONNECTION_BLUETOOTH = "bluetooth"` connection type, which is also a EUI-48 from the same pool of addresses. 

---

**Back to the topic of malformed hardware addresses.**
A problem would arise (that HA currently doesn't seem to account for, and it's not specific to the matter integration) if the device reports a mac address that's not a valid identifier for an individual device. A NULL address, such as `00:00:00:00:00:00`, has the potential to match a bunch of devices. Also, at least half the mac addresses are multicast addresses, also unsuitable for device identification. 

Since this is not matter specific, this should ideally be handled by the "core" HA code.
These checks should be made by the device_registry.